src/main/java/com/icthh/xm/uaa/security/oauth2/idp/jwk/JwkDefinition.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.icthh.xm.uaa.security.oauth2.idp.jwk;
+
+/**
+ * This class copied from {@link org.springframework.security.oauth2.provider.token.store.jwk.JwkDefinition}
+ * and couldn't be imported cause of package private access.
+ * The base representation of a JSON Web Key (JWK).
+ *
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7517">JSON Web Key (JWK)</a>
+ *
+ * @author Joe Grandja
+ * @author Michael Duergner
+ */
+public abstract class JwkDefinition {
```

Comment:
looks like library class, maybe better to add library to gradle that copy paste source code?

Reply:
It won't solve problems with package private access in the class and its inner classes definition.